collectors/hra/collector.py

```diff
@@ -0,0 +1,140 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import logging
+import requests
+import sqlalchemy
+import os
+from datetime import date, timedelta
+import time
+from .. import base
+from .record import Record
+logger = logging.getLogger(__name__)
+
+parse_datetime = base.helpers.parse_datetime
+
+url = 'https://stage.harp.org.uk/HARPApiExternal/api/ResearchSummaries'
+hra_user = os.environ.get('HRA_USERNAME')
+hra_pass = os.environ.get('HRA_PASSWORD')
+
+s = requests.Session()
+
+
+def make_request(url, from_, to, filter=None):
+    # Updates filter is optional-categorical value
+    # 1 - returns all studies for the given period
+    # 2 - returns only studies published in the period
+    # 3 - returns only studies modified in the period
+    result_str = url + '?datePublishedFrom=' + str(from_) + '&datePublishedTo=' + str(to)
+    if filter is None:
+        pass
+    else:
+        result_str = result_str + '&updatesFilter=' + str(filter)
+    response = s.get(result_str, auth=(hra_user, hra_pass))
+    return (response, result_str)
+
+# Module API
+
+
+def collect(conf, conn):
+    errors = 0
+    success = 0
+
+    def parse_response(response, endpoint, from_api, to_date, errors, success, conn=conn):
+        for application in response.json():
+            try:
+                data = {}
+                data['last_updated'] = str(date.today())
```

Comment:
We had discussion about it (I had the same question/proposal) with summary - OKI needs to provide py2/3 support for all open-sourced projects to make it wide deployable. Irony here is anyway we can't do this because `scrapy` and `docker-cloud` are still py2 only.

I have an issue related to it - https://github.com/opentrials/opentrials/issues/110